Content.Server/_White/EmpFlashlight/EmpHitSystem.cs

```diff
@@ -0,0 +1,55 @@
+﻿using Content.Server.Power;
+using Content.Shared.Emp;
+using Robust.Shared.GameObjects;
+using Content.Shared.Weapons.Melee.Events;
+using Content.Server._White.EmpFlashlight;
+using Content.Server.Emp;
+using Robust.Shared.IoC;
+using Content.Shared.Charges.Systems;
+using Content.Shared.Charges.Components;
+
+namespace Content.Server._White.EmpFlashlight;
+
+public sealed class EmpHitSystem : EntitySystem
+{
+    [Dependency] private readonly EntityLookupSystem _lookup = default!;
+    [Dependency] private readonly EmpSystem _emp = default!;
+    [Dependency] private readonly SharedChargesSystem _charges = default!;
+    public override void Initialize()
+    {
+        base.Initialize();
+        SubscribeLocalEvent<EmpOnHitComponent, MeleeHitEvent>(HandleEmpHit);
+    }
+
+    public bool TryEmpHit(EntityUid uid, EmpOnHitComponent comp, MeleeHitEvent args)
+    {
+        LimitedChargesComponent? charges;
+        if (!TryComp<LimitedChargesComponent>(uid, out charges))
```

Comment:
нет смысла выносить определение `charges` вне TryComp блока
```suggestion
        if (!TryComp(uid, out LimitedChargesComponent? charges))
```

Reply:
Solved